src/google/protobuf/compiler/cpp/cpp_helpers.cc

```diff
@@ -176,6 +176,18 @@ string FieldName(const FieldDescriptor* field) {
   return result;
 }
 
+string EnumName(const EnumValueDescriptor* enum_value, bool should_mangle) {
+  string result = enum_value->name();
+  if (should_mangle) {
+    string lower = enum_value->name();
+    LowerString(&lower);
```

Comment:
The style guide actually recommends that all enum values should be defined in capitalized chars. This adds lots of false positives to the current working code.

Also, could you document in which condition we should set true/false for the 2nd parameter of this function?